api/handlers/app_test.go

```diff
@@ -1772,6 +1774,113 @@ var _ = Describe("App", func() {
 			})
 		})
 	})
+
+	Describe("DELETE /v3/apps/:guid/processes/:process/instances/:instance", func() {
+		When("the app has a single instance in the default (web) process", func() {
+			BeforeEach(func() {
+				processRepo.ListProcessesReturns([]repositories.ProcessRecord{
+					{
+						GUID:             "process-1-guid",
+						SpaceGUID:        spaceGUID,
+						AppGUID:          appGUID,
+						Type:             "web",
+						DesiredInstances: 1,
+					},
+				}, nil)
+				req = createHttpRequest("DELETE", "/v3/apps/"+appGUID+"/processes/web/instances/0", nil)
```

Comment:
It is a good practice to have the test call separated in a `JustBeforeEach`. This `JustBeforeEach` can be defined at the top level potentially parametrizing the number of the instance to be deleted. This number can be set in each `When`'s respective `BeforeEach`